libs/application/template-api-modules/src/lib/modules/templates/grindavik-housing-buyout/grindavik-housing-buyout.service.ts

```diff
@@ -0,0 +1,271 @@
+import { Injectable } from '@nestjs/common'
+import { BaseTemplateApiService } from '../../base-template-api.service'
+import { ApplicationTypes } from '@island.is/application/types'
+import { TemplateApiModuleActionProps } from '../../../types'
+
+import {
+  Person,
+  PersonType,
+  SyslumennService,
+} from '@island.is/clients/syslumenn'
+import { NationalRegistryClientService } from '@island.is/clients/national-registry-v2'
+import { getDomicileOnDate } from './grindavik-housing-buyout.utils'
+import { TemplateApiError } from '@island.is/nest/problem'
+import {
+  notEligible,
+  GrindavikHousingBuyoutAnswers,
+  prerequisites,
+} from '@island.is/application/templates/grindavik-housing-buyout'
+
+import { Fasteign, FasteignirApi } from '@island.is/clients/assets'
+import { isRunningOnEnvironment } from '@island.is/shared/utils'
+
+type CheckResidence = {
+  realEstateId: string
+  realEstateAddress: string
+}
+
+@Injectable()
+export class GrindavikHousingBuyoutService extends BaseTemplateApiService {
+  constructor(
+    private readonly syslumennService: SyslumennService,
+    private nationalRegistryApi: NationalRegistryClientService,
+    private propertiesApi: FasteignirApi,
+  ) {
+    super(ApplicationTypes.GRINDAVIK_HOUSING_BUYOUT)
+  }
+
+  async submitApplication({ application, auth }: TemplateApiModuleActionProps) {
+    const answers = application.answers as GrindavikHousingBuyoutAnswers
+
+    const applicant: Person = {
+      name: answers.applicant.name,
+      ssn: answers.applicant.nationalId,
+      phoneNumber: answers.applicant.phoneNumber ?? '',
+      email: answers.applicant.email ?? '',
+      homeAddress: answers.applicant.address ?? '',
+      postalCode: answers.applicant.postalCode ?? '',
+      city: answers.applicant.city ?? '',
+      signed: false,
+      type: PersonType.Plaintiff,
+    }
+
+    const extraData: { [key: string]: string } = {
+      applicationId: application.id,
+    }
+
+    const uploadDataName = 'Umsókn um kaup á íbúðarhúsnæði í Grindavík'
+    const uploadDataId = 'grindavik-umsokn-1'
+
+    const response = await this.syslumennService.uploadData(
+      [applicant],
+      [],
+      extraData,
+      uploadDataName,
+      uploadDataId,
+    )
+    return { ...response, applicationId: application.id }
+  }
+
+  async checkResidence({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<CheckResidence> {
+    const data = await this.nationalRegistryApi.getResidenceHistory(
+      auth.nationalId,
+    )
+
+    // gervimaður færeyjar pass through
+    if (
+      (isRunningOnEnvironment('local') || isRunningOnEnvironment('dev')) &&
+      auth.nationalId === '0101302399'
+    ) {
+      return {
+        realEstateId: 'F12345',
+        realEstateAddress: 'Vesturhóp 34, 240 Grindavík',
+      }
+    }
+
+    const dateInQuestion = '2023-10-11'
+    const domicileOn10nov = getDomicileOnDate(data, dateInQuestion)
```

Comment:
The date you are checking is October 11 but the variable name suggests you want to check november 10.

Reply:
Good catch